api/serializers/people_serializers.py

```diff
@@ -0,0 +1,11 @@
+from rest_framework import serializers
+from api.models import people
+
+
+class PersonalInfoSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = people.PersonalInfo
+        fields = ['id', 'date_and_time_added', 'first_name', 'last_name', 'tg_username',
+                  'email', 'phone', 'tz_summer_relative_to_utc', 'tz_winter_relative_to_utc',
```

Comment:
> > I think 404 response is not a good idea for our purposes. Imho 404 response is an error. In our case if user doesn't exist this isn't an error, this is a normal situation.
> 
> @aakhterov Well that's the thing: if you **count** the users then yes, it is correct that `0` is a normal result and requires a 200 response. But the main purpose of the view is to tell the bot whether the user **exists or not**. So the most obvious (to me, and maybe to an outside observer) way to go is to try and request the user with the given params, just like you request a web page. And you either get it and get the content with 200 response, or you get no page or user and get a 404 response. To me this looks like a very standard and predictable scenario for a case of requesting one resource that may or may not exist. That's why I thought that counting is a bit confusing solution for the task we have here.

I got you logic. My logic is above. We can do as you describe, we can do as I've done. More over we can invent more variants and they also will be appropriate.  Also we can find pro and cons in all variants. If to be honest, I don't see a sense to talk about it endlessly.

Reply:
You can do as you wish, as long as it works and as long as an outside developer understands why you did this to solve a problem of existence of an object.

Reply:
> You can do as you wish, as long as it works and as long as an outside developer understands why you did this to solve a problem of existence of an object. Right now I'm sure they would have same questions as I do ("why count to check for existence").

Because, it exists in the documentation. The outside developer also maybe surprise, why do we return an entiry objects with some fields in response. And why does he or she get an error in response. It depends on developer.

Reply:
It actually depends on our task. Maybe the bot will want to show something informative to the user. Or the backend just gets the entire object, but returns only `True` or `False` to the bot.

At any case, we both agree that our task is to provide the bot with the information whether the user **exists**. At least we should make the backend return `True` or `False` instead of number; and change the url:

```
urlpatterns = [
    path('users/count/', people.UserCountView.as_view()),
]
```
The bot shouldn't need to send a request to an url with `/count/` when all it needs is a check for existence. What happens "under the hood" - you're absolutely right - is the backend's choice (counting, getting object or 404, whatever). But the bot must think that it is simply asking for existence.

Reply:
Ok we are in a circle. I don't want to argue

Reply:
@lemontree210 , I don't want to argue constantly. I waste more time for disputes then for coding. If you don't agree with my point of view - no problems - please, design endpoints, I'll implement them and we'll forget about this.

Reply:
@aakhterov To me, that's not how teamwork works... We present logical arguments and come to a mutually agreeable solution. My argument is: the bot only need to know about the existence of an object. Why should it send a request to a `/count/` url? How does that enforce principles of loose coupling and other principles we both agree are important?

Reply:
My logic is the following: We have 3 roughly identical options. 
**1) Return an entire object.** 
Pro: 
- unversality (can uses somewhere else)
Cons:
- it's not obviously which fields do we return (only from PersonalInfo or PersonalInfo + data from joined tabels)
- There are can be more then an one object with parameters from request. Hence we need to return list and add pagination in the response. And all this is unnecessary inforamtion for this request.  

**2) Return only boolean value**
Pro: 
- return only what we need in this request
Cons:
- return only that we need in this request (this option is also cons because it makes this endpoint very specific, we won't be able to use it somewhere else)
- in this note [https://criteria.sh/blog/returning-booleans-in-responses](https://criteria.sh/blog/returning-booleans-in-responses) talks about something similar. I understand that it isn't exactly our case, but they don't recommend to use boolean value in response. Instead of this they offer "change the Boolean value to an Enumeration".

**3) Return something in between** - not so big as list of objects, but not so specific as only boolean value. For example, objects count
Pro: 
- more universal than option #2
- no unnecessary data   
Cons:
- less obvious than option #2
- less universal than option #1 

> To me, that's not how teamwork works...

I've chosen option #3, but in general I think all this three options are identical. Hence, I don't want to spend so much time arguing about the decision that isn't fundamental. As for me, this isn't teamwork object. Espessialy when we have many other things to do. 

> Why should it send a request to a `/count/` url?

Because if you are an outside developer and you need to check user existing, you look at the API descriptions and you see an endpoint that returns users count and I think you'll understand you can use it for your purpose.

Reply:
Thank you for the link on boolean values in REST APIs, I agree that simply returning `True` (and nothing else) in a response isn't a good idea.